src/foundry/foundry.js/canvas.d.ts

```diff
@@ -17,401 +20,498 @@
+     * @defaultValue `0`
+     */
+    protected _zoomTime: number;
+
+    /**
+     * Track the last automatic pan time to throttle
+     * @defaultValue `0`
+     */
+    protected _panTime: number;
 
-  notes: NotesLayer | undefined;
+    /**
+     * An object of data which is temporarily cached to be reloaded after the canvas is drawn
+     * @defaultValue `{}`
+     */
+    protected _reload: Partial<{ scene: string; layer: string; controlledTokens: string[]; targetedTokens: string[] }>;
 
-  tokens: TokenLayer | undefined;
-
-  lighting: LightingLayer | undefined;
-
-  sounds: SoundsLayer | undefined;
-
-  sight: SightLayer | undefined;
-
-  effects: EffectsLayer | undefined;
-
-  controls: ControlsLayer | undefined;
+    /**
+     * A Set of unique pending operation names to ensure operations are only performed once
+     */
+    protected _pendingOperationNames: Set<string>;
+
+    /**
+     * The pixel radius of blur distance that should be applied for the current zoom level
+     * @defaultValue `0`
+     */
+    blurDistance: number;
+
+    /**
+     * An array of blur filter instances which are modified by the zoom level and the "soft shadows" setting
+     * @defaultValue `[]`
+     */
+    blurFilters: PIXI.filters.BlurFilter[];
+
+    /**
+     * The singleton interaction manager instance which handles mouse interaction on the Canvas.
+     */
+    mouseInteractionManager: MouseInteractionManager<Exclude<this['stage'], undefined>> | undefined;
+
+    /**
+     * A reference to the MouseInteractionManager that is currently controlling pointer-based interaction, or null.
+     * @defaultValue `null`
+     */
+    currentMouseManager: MouseInteractionManager<Exclude<this['stage'], undefined>> | null;
+
+    /**
+     * Initialize the Canvas by creating the HTML element and PIXI application.
+     * This step should only ever be performed once per client session.
+     * Subsequent requests to reset the canvas should go through Canvas#draw
+     */
+    initialize(): void;
+
+    app: PIXI.Application | undefined;
+
+    stage: PIXI.Container | undefined;
+
+    protected _dragDrop: DragDrop | undefined;
+
+    msk: PIXI.Graphics | undefined;
+
+    /**
+     * The id of the currently displayed Scene.
+     */
+    get id(): string | null;
+
+    /**
+     * A mapping of named CanvasLayer classes which defines the layers which comprise the Scene.
+     */
+    static get layers(): typeof CONFIG.Canvas.layers;
+
+    /**
+     * An Array of all CanvasLayer instances which are active on the Canvas board
+     */
+    get layers(): CanvasLayer[];
+
+    /**
+     * Return a reference to the active Canvas Layer
+     */
+    get activeLayer(): CanvasLayer | null;
+
+    /**
+     * Create the layers of the game Canvas.
+     * @param stage - The primary canvas stage
+     */
+    protected _createLayers(stage: PIXI.Container): void;
+
+    background: BackgroundLayer | undefined;
```

Comment:
I think these should be moved between `_dragDrop` and `msk`, that's when they are defined in the initialization (by calling`_createLayers`)